docs/datatypes/tset.md

```diff
@@ -0,0 +1,251 @@
+---
+id: datatypes_tset
+title: "TSet"
+---
+
+A `TSet[A]` is transactional data structure built on top of `TMap[A]`.
+
+## Create a TSet
+
+Creating an empty `TSet`:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val emptyTSet: STM[Nothing, TSet[Int]] = TSet.empty[Int]
+```
+
+Or creating a `TSet` with specified values:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val specifiedValuesTSet: STM[Nothing, TSet[Int]] = TSet.make(1, 2, 3)
+```
+
+Or creating a `TSet` from existing `Iterable` structure:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val iterableTSet: STM[Nothing, TSet[Int]] = TSet.fromIterable(List(1, 2, 3))
+```
+
+## Put an element to a `TSet`
+
+Putting new element to a `TSet` extends it if specified element is not already present:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val putElem: UIO[TSet[Int]] = (for {
+  tSet <- TSet.make(1, 2)
+  _    <- tSet.put(3)
+} yield tSet).commit
+```
+
+## Remove an element from a `TSet`
+
+The simplest way to remove an element from `TSet` is using `delete` method:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val deleteElem: UIO[TSet[Int]] = (for {
+  tSet <- TSet.make(1, 2, 3)
+  _    <- tSet.delete(1)
+} yield tSet).commit
+```
+
+Also, it is possible to remove every element that matches predicate function:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val removedEvenElems: UIO[TSet[Int]] = (for {
+  tSet <- TSet.make(1, 2, 3, 4)
+  _    <- tSet.removeIf(_ % 2 == 0)
+} yield tSet).commit
+```
+
+Or you can keep all elements that match predicate function:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val retainedEvenElems: UIO[TSet[Int]] = (for {
+  tSet <- TSet.make(1, 2, 3, 4)
+  _    <- tSet.retainIf(_ % 2 == 0)
+} yield tSet).commit
+```
+
+## Union of a `TSet`
+
+Union of the sets A and B represents the set of distinct element belongs to set A or set B, or both.
```

Comment:
`..the set of elements belonging to ...`. Emphasize which of the sets is modified.